python_modules/dagster/dagster_tests/cli_tests/command_tests/test_telemetry.py

```diff
@@ -67,6 +68,9 @@ def test_dagster_telemetry_enabled(caplog):
                 assert set(message.keys()) == EXPECTED_KEYS
             assert len(caplog.records) == 5
             assert result.exit_code == 0
+        
+        # Needed to avoid file contention issues on windows with the telemetry log file
+        cleanup_telemetry_logger()
```

Comment:
consider some sort of contextmanager here with a descriptive name:

```
@contextmanager
def instance_for_telemetry():
    with instance_for_test(overrides={"telemetry": {"enabled": True}}) as instance:
        yield instance
        cleanup_telemetry_logger()
```
```
with instance_for_telemetry():
     # various assertions
```

Reply:
Will follow up with this.